confmap/provider/s3provider/provider_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3provider
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+// checkURI checks whether the s3-uri is valid
+func checkURI(uri string) error {
+	// check uri's prefix valid or not
+	if !strings.HasPrefix(uri, schemeName+":") {
+		return fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
+	}
+	// Check if users set up their env for S3 Auth check yet
+	if os.Getenv("AWS_ACCESS_KEY_ID") == "" || os.Getenv("AWS_SECRET_ACCESS_KEY") == "" {
+		return fmt.Errorf("unable to fetch access keys for S3 Auth")
+	}
+	// check uri valid or not, should with 'Bucket, Region, Key'
+	_, _, _, err := S3URISplit(uri)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// testRetrieve: Mock how Retrieve() works in normal cases
+type testRetrieve struct{}
+
+func NewTestRetrieve() confmap.Provider {
+	return &testRetrieve{}
+}
+
+func (fp *testRetrieve) Retrieve(ctx context.Context, uri string, watcher confmap.WatcherFunc) (confmap.Retrieved, error) {
+	// check URI
+	err := checkURI(uri)
+	if err != nil {
+		return confmap.Retrieved{}, err
+	}
+	// read local config file and return
+	f, err := ioutil.ReadFile("../../testdata/config.yaml")
```

Comment:
Where is this test data? Same for the other test data used

Reply:
Yeah, actually I am using the test data under opentelemetry-collector/confmap. But seems I should use test data under opentelemetry-collector/confmap/provider/fileprovider/testdata instead. In this way, we can in a way keep consistency with yaml/file/envmapprovider.

I will update this and push.

Reply:
Talked about it offline, but for visibility: I suggest to have your own testdata directory for the s3 provider specifically